runtime/src/bank/partitioned_epoch_rewards/mod.rs

```diff
@@ -39,14 +44,98 @@ pub(crate) struct PartitionedStakeReward {
     pub commission: u8,
 }
 
-type PartitionedStakeRewards = Vec<PartitionedStakeReward>;
+/// A vector of stake rewards.
+#[derive(Debug, Default)]
+pub(crate) struct PartitionedStakeRewards {
+    /// Inner vector.
+    rewards: Vec<Option<PartitionedStakeReward>>,
+    /// Number of `Some` elements.
+    len_some: usize,
+}
+
+impl PartitionedStakeRewards {
+    pub(crate) fn with_capacity(capacity: usize) -> Self {
+        let rewards = Vec::with_capacity(capacity);
+        Self {
+            rewards,
+            len_some: 0,
+        }
+    }
+
+    pub(crate) fn capacity(&self) -> usize {
+        self.rewards.capacity()
+    }
+
+    /// Number of `Some` elements.
+    pub(crate) fn len_some(&self) -> usize {
```

Comment:
This method addresses https://github.com/anza-xyz/agave/pull/6900#pullrequestreview-3158696020

I didn't implement `Deref` and `DerefMut` on purpose - this way, the `len` method from the inner `Vec` is not available, so consumers of `PartitionedStakeRewards` are forced to use `len_some`.